nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/log/kafka/KafkaLogTailer.java

```diff
@@ -343,6 +355,20 @@ public void commit() {
         }
     }
 
+    /**
+     * Commits the consumer at its current position regardless of lastOffsets or lastCommittedOffsets
+     */
+    protected void forceCommit() {
+        log.info("Force commit after a move");
+        final Map<TopicPartition, OffsetAndMetadata> offsets = new HashMap<>(topicPartitions.size());
+        topicPartitions.forEach(topicPartition -> offsets.put(topicPartition,
+                new OffsetAndMetadata(consumer.position(topicPartition))));
```

Comment:
You can probably use a collector to create `offsets`, something like

    Map<TopicPartition, OffsetAndMetadata> offsets = topicPartitions.stream().collect(Collectors.toMap(
        tp -> tp, tp -> new OffsetAndMetadata(consumer.position(tp))))

You can `import static` the `Collectors.toMap` for brevity.